File: Data Structures & Algorithms/number-of-one-bits/submission-1.java
class Solution {
    public int hammingWeight(int n) {
     int count = 0;
     int temp = n & 1;
     while(n != 0){
        if (temp == 1){
            count += 1;
           n = n >>1;
           temp = n & 1;
        }
        else{
            n = n>>1;
            temp = n & 1;
        }
     }
     return count;
}
}

File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        output = []

        for i in range(n+1):
            output.append(self.countbit(i))

        return output 

    def countbit(self,n):
        count = 0
        while n > 0:
            count += n & 1
            n = n >> 1
        return count